app/src/gui/components/notebook/record_table.tsx

```diff
@@ -358,16 +345,93 @@ function RecordsTable(props: RecordsTableProps) {
         },
       ];
 
+  // Fetch the current user when the component mounts
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const userId = await getCurrentUserId(project_id);
+        setCurrentUser(userId);
+      } catch (error) {
+        console.error('Error fetching user ID:', error);
+      }
+    };
+    fetchUser();
+  }, [project_id]);
```

Comment:
This pattern is best avoided 

a) we should have a context provider for current user
b) the useState -> useEffect combo for doing async is an anti-pattern in a codebase using React Query. I'd recommend a very simple custom hook useGetCurrentUser which takes project ID as prop, and returns the result of a useQuery doing what you've done above- then we get error, loading, caching etc all for free!

Too much to do context provider for this PR - so useQuery would be best.

Reply:
Yes ofcouse,
fixed this